visualization/income_edu/income.html

```diff
@@ -255,13 +256,62 @@
 				.attr("transform", "translate(" + margin + ",0)")
 				.call(wage_axis);
       
-			// After the axis get rendered, further customize the x axis ticks and labels:
-			// Make labeled ticks longer
-			d3.selectAll("g.x.axis g.tick line")
-				.attr("y2", (x, i) => major_tick[i] ? 15 : 6);
-			// Move labels lower to match the longer ticks
-			d3.selectAll("g.x.axis g.tick text")
-				.attr("y", (x, i) => major_tick[i] ? 20 : 11)
+			// After the axis get rendered, move x-axis labels out of the SVG and into divs in the #chart container
+			// Delete the SVG labels
+			d3.selectAll("g.x.axis g.tick text").remove();
+			
+			// Create the #chart div labels
+			x_labels = chart_div.selectAll('.x-label')
+				.data(edu_labels);
+			x_labels = x_labels.enter()
+				.append("div")
+				.attr("class", "x-label")
+				.text(x => x);
+			
+			// Adjust their width based on the distance between ticks
+			var min_label_width = 55,
+				tick_distance = 0.9*(edu_scale(2)-edu_scale(1)),
+				label_width;
+
+			if (tick_distance >= min_label_width) { 
```

Comment:
Consider refactoring these into a function, they're quite long and not esp. easy to read